ONSdigital/eq-author-app
# Ear 327 make share a page

> BEFORE MAKING YOUR PR... There must be no linting errors and all tests must pass

### What is the context of this PR?

Converted the share modal to a share page.

Mostly styling changes and a little bit of modification to the functionality of the user search. Involves the inclusion of a submit button, instead of just adding users from the dropdown (apparently called a downshift)

In this ticket I extracted the `styled-components` I used into separate files and included the prop-types at the top of the file to give a better indication of the requirements of the component.

Feedback would be appreciated if it's worth doing this in the future or not.

Feedback for the tests would also be appreciated. In terms of whether there's enough or they hit the right stuff. Felt a bit rusty writing the tests and a cheat sheet would have been decent. Things like mocking the graphql queries.

The coverage of the tests isn't great because I haven't tested the query wrappers. Looks bad because they have a few if loops in them but not sure if they should be tested?

### How to review 

1. Create a questionnaire
2. Click Sharing - and be amazed!

3. Hit the shareable link
   - *locally it should look something like `/launch/:questionnaireid`*
   - **Toast** should appear at the bottom of the screen saying 'link copied to clipboard'

4. Toggle isPublic
    - Toggle switch should change from on/off
    - a network request should be made with the correct value
    - can export the questionnaire and check if `questionnaire.isPublic` has updated

5. Adding editors
    - hard to test locally due to lack of users
    - if you are the owner, shouldn't be able to add your self
    - the owner should be highlighted in black
    - if a user can be added, can't add copies of the same user
    - when the close button is clicked - the user is removed

Things that might be hard to check
- User doesn't have name and so is replaced with email
- Adding many users doesn't bust out of the page container

### What to do after everything is green

1. SQUASH all the commits!
